docs/src/pages/styles/advanced/advanced.md

```diff
@@ -552,3 +552,207 @@ You can also combine JSS generated class names with global ones.
 
 JSS uses feature detection to apply the correct prefixes.
 [Don't be surprised](https://github.com/mui-org/material-ui/issues/9293) if you can't see a specific prefix in the latest version of Chrome. Your browser probably doesn't need it.
+
+## TypeScript usage
+
+Using `withStyles` in TypeScript can be a little tricky, but there are some utilities to make the experience as painless as possible.
+
+### Using `createStyles` to defeat type widening
+
+A frequent source of confusion is TypeScript's [type widening](https://mariusschulz.com/blog/typescript-2-1-literal-type-widening), which causes this example not to work as expected:
+
+```ts
+const styles = {
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+};
+
+withStyles(styles);
+//         ^^^^^^
+//         Types of property 'flexDirection' are incompatible.
+//           Type 'string' is not assignable to type '"-moz-initial" | "inherit" | "initial" | "revert" | "unset" | "column" | "column-reverse" | "row"...'.
+```
+
+The problem is that the type of the `flexDirection` prop is inferred as `string`, which is too arbitrary. To fix this, you can pass the styles object directly to `withStyles`:
```

Comment:
"arbitrary" doesn't seem like the right word here – in essence it means "random".